sqlglot/optimizer/qualify_columns.py

```diff
@@ -358,18 +382,26 @@ def _expand_stars(
     replace_columns: t.Dict[int, t.Dict[str, str]] = {}
     coalesced_columns = set()
 
-    # TODO: handle optimization of multiple PIVOTs (and possibly UNPIVOTs) in the future
     pivot_columns = None
     pivot_output_columns = None
-    pivot = t.cast(t.Optional[exp.Pivot], seq_get(scope.pivots, 0))
+    pivot_exclude_columns = None
 
-    has_pivoted_source = pivot and not pivot.args.get("unpivot")
-    if pivot and has_pivoted_source:
-        pivot_columns = set(col.output_name for col in pivot.find_all(exp.Column))
+    pivot = t.cast(t.Optional[exp.Pivot], seq_get(scope.pivots, 0))
```

Comment:
can you have multiple pivots? why not just do a for loop?

Reply:
You can but even if we do the for loop it won't work correctly; the semantics are tricky to get right because e.g. each pivot feeds the columns it produces into the next, one so it becomes messy when there are more than one...

I'd just leave it for now and look into this when there's more demand, chained pivots should be rare anyways.